tools/test_backups2datalad.py

```diff
@@ -191,3 +196,25 @@ def test_custom_commit_date(tmp_path: Path) -> None:
         ds.save(message="Add a file")
     ts = readcmd("git", "show", "-s", "--format=%aI", "HEAD", cwd=tmp_path)
     assert ts == "2021-06-01T12:34:56+00:00"
+
+
+def test_custom_commit_author(tmp_path: Path) -> None:
```

Comment:
a dedicated unittest for the helper is nice, but what would be more valuable IMHO is to extend `test_1` with assertions that commits for changes, tags for Published versions have a custom AUTHOR

Reply:
@yarikoptic The creator of a tag (not to be confused with the creator of the commit a tag points to) is set via `GIT_COMMITTER_{NAME,EMAIL}`, not via `GIT_AUTHOR_*`.  Should that still be tested for?

Reply:
@yarikoptic Also, if `GIT_AUTHOR_{NAME,EMAIL}` are only set when setting `GIT_AUTHOR_DATE`, then the dataset's initial commits as well as any commits that download assets from S3 will have their author set to the configured local user instead of "DANDI User".

Reply:
> @yarikoptic The creator of a tag (not to be confused with the creator of the commit a tag points to) is set via `GIT_COMMITTER_{NAME,EMAIL}`, not via `GIT_AUTHOR_*`. Should that still be tested for?

thanks for the explanation. Yes - let's set COMMITTER name/date then to be the one for the AUTHOR (i.e. from archive).  It would be great to have it tested, yes.

> @yarikoptic Also, if `GIT_AUTHOR_{NAME,EMAIL}` are only set when setting `GIT_AUTHOR_DATE`, then the dataset's initial commits as well as any commits that download assets from S3 will have their author set to the configured local user instead of "DANDI User".

Since we know created date, and IIRC we commit assets for a specific state of the draft (hence known modified date), all those activities should just get the same context helper which would set `GIT_AUTHOR_{NAME,EMAIL,DATE}`, or what am I missing? 
I think that the only committed activity which should go without tuning AUTHOR is the tag merging back into main branch - merged #69 .

Reply:
@yarikoptic 

> let's set COMMITTER name/date then to be the one for the AUTHOR

Do you mean that the commiter should be set to "DANDI User" only for tagging or also for commits?

As for the other part, do you mean that *all* commits to the dataset while syncing (including the call to `ds.create()`, if any) should use the same date?

Reply:
> > let's set COMMITTER name/date then to be the one for the AUTHOR
> 
> Do you mean that the commiter should be set to "DANDI User" only for tagging or also for commits?

yes, for the commits which reflect the state of the dandiset (draft or another version) with a known `modified` date from the archive.

> As for the other part, do you mean that _all_ commits to the dataset while syncing (including the call to `ds.create()`, if any) should use the same date?

Not exactly: For `ds.create()` use the `.created` date (not `modified`).  `ds.create` creates two commits (2nd one to tune .gitattributes in `cfg_text2git` procedure) but that is IMHO an ok price to pay.  For the others -- yes, unless it is some home keeping activity (merge of a tag) for which there were no `.modifed` date coming from the archive

Reply:
and in the last portion of above -- I meant only `AUTHOR` date, not COMMITTER -- those would be "real clock"

Reply:
@yarikoptic So, to summarize:

| Action | Set author? | Set committer? | Author date | Commit date |
| --- | --- | --- | --- | --- |
| `ds.create()` | Yes | No | `dandiset.version.created` | Default |
| Commit at end of `sync_dataset()` | Yes | Yes | `dandiset.version.modified` | Default |
| "[DATALAD] Download URLs" commit (`ds.download_url()`?) | Yes | Yes | `dandiset.version.modified` | Default |
| Commit updating just `dandiset.yaml` when creating a release tag | Yes | Yes | `dandiset.version.created` | Default |
| Creating release tag | N/A | Yes | N/A | `dandiset.version.created` |

Is this what you want?

Reply:
More of:

| Action | Set author? | Set committer? | Author date | Commit date |
| --- | --- | --- | --- | --- |
| `ds.create()` | Yes | No | `dandiset.version.created` | Default |
| Commit at end of `sync_dataset()` | Yes | No | `dandiset.version.modified` | Default |
| "[DATALAD] Download URLs" commit (`ds.download_url()`?) | Yes | No | `dandiset.version.modified` | Default |
| Commit updating just `dandiset.yaml` when creating a release tag | Yes | No | `dandiset.version.created` | Default |
| Creating release tag | N/A | Yes | N/A | `dandiset.version.created` (note: make _NAME/EMAIL as used for the AUTHOR) |

i.e. we simply default to git configuration to provide committer generally, and only set it (with information from archive) for the tag since there is no _AUTHOR for tags as you mentioned

Reply:
@yarikoptic Updates made.